### ISSUE TITLE
Add a flag AutoMeasurement.Initialized

### ISSUE THREAD

**Issue by xmedeko** [AttackPattern/CSharpAnalytics#35]
I am using `AutoMeasurement` class from a non-gui library and it need to know, if the `AutoMeasurement` is initialized or not. (`AutoMeasurement.Instance` throws an exception.) Some flag like

``` csharp
public static bool Initialized
{
    get { return _instance != null; }
}
```

**Comment by JimiC** [comment]
Before calling any method of AutoMeasurement, you **_always**_ have to declare an instance first. i.e.
`AutoMeasurement.Instance = new WinFormAutoMeasurement();`

@damieng This issue can be closed.

**Comment by xmedeko** [comment]
I have done a log4net Appender which sends errors to GA by `CSharpAnalytics`. I have to check if the `AutoMeasurement` is initialized or not. So far, I have implemented my own flag, but it would be nice to have such flag build in the library itself, so the code would not rely on any external flags.

**Comment by JimiC** [comment]
Wouldn't code like `AutoMeasurement.Instance != null` suffice?

Edit: Upon checking the above proposed code, I realized that it throws an exception.

**Comment by damieng** [comment]
The problem with this is it means now every single call to AutoMeasurement has to be wrapped in the null check. The intent was you could just call it and it would do it when it is finished initializing like a queue. (although obviously there are some race conditions right now)

**Comment by JimiC** [comment]
The other way around is to encapsulate the null check of the Instance in a try/catch and use a self flag as @xmedeko might do.

**Comment by xmedeko** [comment]
@damieng Yep, this log4net appender has wrap every call to the initialized check. But it is a special pieace fo code. The rest of the application does not check for initialized before `AutoMeasurement.Instance` use. @JimiC try/catch is IMHO to costly comparing to the null check.

Another solution instead of `Initialized` may be

``` c#
public static BaseAutoMeasurement InstanceOrNull { get { return _instance; } }
```

Which is a hint to developer that `Instance` is a normal usage and `InstanceOrNull` is an exceptional usage. So they would not blindly wrap every call to `Initialized` check when it is not necessary.
